src/lib/swapper/LifiSwapper/LifiSwapper.ts

```diff
@@ -72,25 +75,27 @@ export class LifiSwapper implements Swapper<EvmChainId> {
   /**
    * Builds a trade with definitive rate & txData that can be executed with executeTrade
    **/
-  async buildTrade(input: BuildTradeInput): Promise<LifiTrade> {
+  async buildTrade(input: BuildTradeInput): Promise<Result<LifiTrade, SwapErrorRight>> {
     return await buildTrade(input, this.lifiAssetMap, this.lifiChainMap)
   }
 
   /**
    * Get a trade quote
    */
-  async getTradeQuote(input: GetEvmTradeQuoteInput): Promise<LifiTradeQuote> {
+  async getTradeQuote(
+    input: GetEvmTradeQuoteInput,
+  ): Promise<Result<LifiTradeQuote, SwapErrorRight>> {
     const minimumCryptoHuman = getMinimumCryptoHuman(input.sellAsset)
     const minimumSellAmountBaseUnit = toBaseUnit(minimumCryptoHuman, input.sellAsset.precision)
     const isBelowMinSellAmount = bnOrZero(input.sellAmountBeforeFeesCryptoBaseUnit).lt(
       minimumSellAmountBaseUnit,
     )
```

Comment:
all of this needs to be replaced to take advantage of gonads, but can totally be done after this PR gets merged

Reply:
Absolutely, this PR purposely doesn't change the flow yet, which is why it only contributes to, but doesn't close https://github.com/shapeshift/web/issues/4237

Reply:
Resolving this thread, as we can tackle this in a future PR 👌